go/internal/flag/flag_test.go

```diff
@@ -0,0 +1,308 @@
+	oldPflagCommandLine := pflag.CommandLine
+	oldCommandLine := goflag.CommandLine
+
+	defer func() {
+		os.Args = oldOsArgs
+		pflag.CommandLine = oldPflagCommandLine
+		goflag.CommandLine = oldCommandLine
+	}()
+
+	pflag.CommandLine = pflag.NewFlagSet("testFlagSet", pflag.ExitOnError)
+
+	os.Args = []string{
+		"-test.run",
+		"TestFilter",
+		"otherArgs1",
+		"otherArgs2",
+		"-test.run=TestFilter",
+	}
+
+	ParseFlagsForTest()
+
+	expectedOsArgs := []string{
+		"otherArgs1",
+		"otherArgs2",
+	}
+
+	assert.Equal(t, expectedOsArgs, os.Args)
+	assert.Equal(t, true, pflag.Parsed())
+}
+
+func TestParsed(t *testing.T) {
+	oldPflagCommandLine := pflag.CommandLine
+	oldCommandLine := goflag.CommandLine
+
+	defer func() {
+		pflag.CommandLine = oldPflagCommandLine
+		goflag.CommandLine = oldCommandLine
+	}()
+
+	pflag.CommandLine = pflag.NewFlagSet("testPflagSet", pflag.ExitOnError)
+	goflag.CommandLine = goflag.NewFlagSet("testGoflagSet", goflag.ExitOnError)
+
+	b := Parsed()
+	assert.False(t, b)
+
+	pflag.Parse()
+	b = Parsed()
+	assert.True(t, b)
+}
+
+func TestLookup(t *testing.T) {
+	oldPflagCommandLine := pflag.CommandLine
+	oldCommandLine := goflag.CommandLine
+
+	defer func() {
+		pflag.CommandLine = oldPflagCommandLine
+		goflag.CommandLine = oldCommandLine
+	}()
+
+	pflag.CommandLine = pflag.NewFlagSet("testPflagSet", pflag.ExitOnError)
+	goflag.CommandLine = goflag.NewFlagSet("testGoflagSet", goflag.ExitOnError)
+
+	var testGoFlag, testPflag, testFlag bool
+
+	goflag.BoolVar(&testGoFlag, "testGoFlag", true, "")
+	goflag.BoolVar(&testFlag, "t", true, "")
+	pflag.BoolVar(&testPflag, "testPflag", true, "")
+
+	testCases := []struct {
+		shorthand string
+		name      string
+	}{
+		{
+			// If single character flag is passed, the shorthand should be the same
+			shorthand: "t",
+			name:      "t",
+		},
+		{
+			shorthand: "",
+			name:      "testGoFlag",
+		},
+		{
+			shorthand: "",
+			name:      "testPflag",
+		},
+	}
+
+	for _, tt := range testCases {
+		f := Lookup(tt.name)
+
+		assert.NotNil(t, f)
+		assert.Equal(t, tt.shorthand, f.Shorthand)
+		assert.Equal(t, tt.name, f.Name)
+	}
+
+	f := Lookup("non-existent-flag")
+	assert.Nil(t, f)
+}
+
+func TestArgs(t *testing.T) {
+	oldPflagCommandLine := pflag.CommandLine
+	oldOsArgs := os.Args
+
+	defer func() {
+		pflag.CommandLine = oldPflagCommandLine
+		os.Args = oldOsArgs
+	}()
+
+	pflag.CommandLine = pflag.NewFlagSet("testPflagSet", pflag.ExitOnError)
+
+	os.Args = []string{
+		"arg0",
+		"arg1",
+		"arg2",
+		"arg3",
+	}
+
+	expectedArgs := []string{
+		"arg1",
+		"arg2",
+		"arg3",
+	}
+
+	pflag.Parse()
+	// Should work equivalent to pflag.Args if there's no double dash
+	args := Args()
+	assert.Equal(t, expectedArgs, args)
+
+	arg := Arg(2)
+	assert.Equal(t, "arg3", arg)
+
+	// Should return empty string if the index is greater than len of CommandLine.args
+	arg = Arg(3)
+	assert.Equal(t, "", arg)
+}
+
+func TestIsZeroValue(t *testing.T) {
+	oldCommandLine := goflag.CommandLine
```

Comment:
again, we don't need to mutate global state for this test

Reply:
Alternatively, created new `FlagSet`